### PR TITLE
Add recipe for opensubdiv

### PR DESCRIPTION
**Pull request by traversaro**
[opensubdiv](https://github.com/PixarAnimationStudios/OpenSubdiv) is an Open-Source subdivision surface library written in C++ from Pixar.

The package name (`libopensubdiv`) contains the `lib` prefix as requested in https://github.com/conda-forge/staged-recipes/pull/19764#pullrequestreview-1085836936 for C/C++ libraries. The feedstock is named `opensubdiv` to match the repo name and support different package outputs in the future, if that turns out to be necessary.

I am adding opensubdiv as it is a dependency for the usd library (https://github.com/PixarAnimationStudios/OpenUSD), that itself is a dependency of the mujoco library, see https://github.com/conda-forge/mujoco-feedstock/pull/62#issuecomment-2396137162 .

The package contains two patches, one to disable the installation of static libraries to be compliant with [CFEP-03](https://github.com/conda-forge/cfep/blob/main/cfep-03.md), and another to support compilation of shared libraries on Windows (see https://github.com/PixarAnimationStudios/OpenSubdiv/issues/71#issuecomment-2396920840).

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge/staged-recipes @conda-forge/help-c-cpp The PR is ready for review, thanks!

File: recipes/opensubdiv/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - if not exist %PREFIX%\\Library\\lib\\osdCPU_import.lib exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\bin\\osdCPU.dll exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\lib\\cmake\\OpenSubdiv\\OpenSubdivConfig.cmake exit 1  # [win]
+
+about:
+  home: https://github.com/PixarAnimationStudios/OpenSubdiv
+  # Pixar is an actual SPDX license : https://spdx.org/licenses/Pixar.html
+  # However, for some reason the linter complains: https://github.com/conda-forge/staged-recipes/pull/27800#issuecomment-2395512387
+  # So as a workaround we mark it as LicenseRef-Pixar
+  license: LicenseRef-Pixar
```
Comment (xhochy) [inline recipes/opensubdiv/meta.yaml:50]: https://github.com/conda-forge/license-expression-feedstock/pull/8 should solve it.
```suggestion
  license: Pixar
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by xhochy** [comment]
I can take care of the license later today.

**Comment by traversaro** [comment]
> I can take care of the license later today.

Thanks! I wanted to open a conda-smithy issue but I forgot to do so, my bad.

**Review by xhochy — APPROVED** [review]

**Comment by xhochy** [comment]
@conda-forge-admin please relint

**Pull request merged**